### DISCUSSION TITLE
Problems with installing Monailabel

### DISCUSSION THREAD

**Discussion by ylim99** [Project-MONAI/MONAILabel#1538] [Q&A]
Collecting numpymaxflow==0.0.5 (from monailabel)
  Using cached numpymaxflow-0.0.5.tar.gz (14 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'error'

 error: subprocess-exited-with-error
  
  pip subprocess to install build dependencies did not run successfully.
  exit code: 1
  
  [6 lines of output]
  Collecting setuptools
    Obtaining dependency information for setuptools from https://files.pythonhosted.org/packages/4f/ab/0bcfebdfc3bfa8554b2b2c97a555569c4c1ebc74ea288741ea8326c51906/setuptools-68.1.2-py3-none-any.whl.metadata
    Using cached setuptools-68.1.2-py3-none-any.whl.metadata (6.2 kB)
  ERROR: Ignored the following versions that require a different python version: 1.21.2 Requires-Python >=3.7,<3.11; 1.21.3 Requires-Python >=3.7,<3.11; 1.21.4 Requires-Python >=3.7,<3.11; 1.21.5 Requires-Python >=3.7,<3.11; 1.21.6 Requires-Python >=3.7,<3.11
  ERROR: Could not find a version that satisfies the requirement numpy==1.21.6 (from versions: 1.3.0, 1.4.1, 1.5.0, 1.5.1, 1.6.0, 1.6.1, 1.6.2, 1.7.0, 1.7.1, 1.7.2, 1.8.0, 1.8.1, 1.8.2, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.10.0.post2, 1.10.1, 1.10.2, 1.10.4, 1.11.0, 1.11.1, 1.11.2, 1.11.3, 1.12.0, 1.12.1, 1.13.0, 1.13.1, 1.13.3, 1.14.0, 1.14.1, 1.14.2, 1.14.3, 1.14.4, 1.14.5, 1.14.6, 1.15.0, 1.15.1, 1.15.2, 1.15.3, 1.15.4, 1.16.0, 1.16.1, 1.16.2, 1.16.3, 1.16.4, 1.16.5, 1.16.6, 1.17.0, 1.17.1, 1.17.2, 1.17.3, 1.17.4, 1.17.5, 1.18.0, 1.18.1, 1.18.2, 1.18.3, 1.18.4, 1.18.5, 1.19.0, 1.19.1, 1.19.2, 1.19.3, 1.19.4, 1.19.5, 1.20.0, 1.20.1, 1.20.2, 1.20.3, 1.21.0, 1.21.1, 1.22.0, 1.22.1, 1.22.2, 1.22.3, 1.22.4, 1.23.0rc1, 1.23.0rc2, 1.23.0rc3, 1.23.0, 1.23.1, 1.23.2, 1.23.3, 1.23.4, 1.23.5, 1.24.0rc1, 1.24.0rc2, 1.24.0, 1.24.1, 1.24.2, 1.24.3, 1.24.4, 1.25.0rc1, 1.25.0, 1.25.1, 1.25.2, 1.26.0b1)
  ERROR: No matching distribution found for numpy==1.21.6
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

pip subprocess to install build dependencies did not run successfully.
exit code: 1

**Comment by SachidanandAlle** [comment]
i guess you are working on python 3.11?  MONAI Label is not yet verified on 3.11+
however this can be fixed and verified (by upgrading few dependencies for 3.11 like numpy etc..)

if you can, please try to fix this on 3.11 and raise a PR to support python 3.11+

**Comment by elitap** [comment]
As I also run into this recently. I was able to get monailabel installed on 3.11 by simply manually installing numpymaxflow beforehand. For some reason it useses the requirements-dev which requires an exact numpy version (i.e. 1.21.6) whereas the requirments.txt from the numpymaxflow repo (https://github.com/masadcv/numpymaxflow) is a bit more flexible (>= 1.21.6). So just follow the steps in the repo and install monailable after and it will run on 3.11.

Maybe this helps somebody else with the same problem :-)

**Reply by miritrope** [reply]
I followed the instructions exactly as you provided, and it functioned successfully!

**Comment by SachidanandAlle** [comment]
thanks... can u please raise a PR if that can help to fix the problem on 3.11 (you can add if 3.11 conditions in the requirements.txt)

**Comment by elitap** [comment]
Hm the problem is numpymaxflow which uses a too old numpy version for python 3.11... I played around with it a bit, but without manually installing it from source I couldn't get it installed on 3.11. I created a PR there updating the numpy version. Once it its accepted monailable should install on 3.11 without any changes required here...

**Reply by empat6** [reply]
i tried to install numpymaxflow in python 3.11 but it cannot be installed. Do you found another solution? or should i use older version of python to use monailabel?

**Reply by diazandr3s** [reply]
I think monailabel is not verified over Python 3.11 version. Please use 3.9 or 3.10

**Reply by elitap** [reply]
Sure I would also always suggest using the officially supported python versions. If you need to stick with 3.11 like me, in order to get numpymaxflow running, install from source as described in their github repo. Imoprtantly use their requirements.txt and NOT the requirements-dev.txt to install the numpy dependency. For some reoson they created their pipy builds with the dev version, which uses an old numpy version without python 3.11 support. But long story short, if they accept the PR and update to a newer numpy or at least use a requirements.txt  file allowing newer numpy version, this should resolve by itself soon :-)
